atc/syslog/drainer.go

```diff
@@ -49,51 +50,66 @@ func (d *drainer) Run(ctx context.Context) error {
 			logger.Error("Syslog drainer connecting to server error.", err)
 			return err
 		}
-		defer syslog.Close()
+
+		// ignore any errors coming from syslog.Close()
+		defer db.Close(syslog)
 
 		for _, build := range builds {
-			events, err := build.Events(0)
+			err := d.drainBuild(logger, build, syslog)
 			if err != nil {
-				logger.Error("Syslog drainer getting build events error.", err)
 				return err
 			}
+		}
+	}
+	return nil
+}
 
-			for {
-				ev, err := events.Next()
-				if err != nil {
-					if err == db.ErrEndOfBuildEventStream {
-						break
-					}
-					logger.Error("Syslog drainer getting next event error.", err)
-					return err
-				}
-
-				if ev.Event == "log" {
-					var log event.Log
-
-					err := json.Unmarshal(*ev.Data, &log)
-					if err != nil {
-						logger.Error("Syslog drainer unmarshalling log error.", err)
-						return err
-					}
-
-					payload := log.Payload
-					tag := build.TeamName() + "/" + build.PipelineName() + "/" + build.JobName() + "/" + build.Name() + "/" + string(log.Origin.ID)
-
-					err = syslog.Write(d.hostname, tag, time.Unix(log.Time, 0), payload)
-					if err != nil {
-						logger.Error("Syslog drainer sending to server error.", err)
-						return err
-					}
-				}
+func (d *drainer) drainBuild(logger lager.Logger, build db.Build, syslog *Syslog) error {
+	events, err := build.Events(0)
+	if err != nil {
+		logger.Error("Syslog drainer getting build events error.", err)
+		return err
+	}
+
+	// ignore any errors coming from events.Close()
+	defer db.Close(events)
+
+	for {
+		ev, err := events.Next()
+		if err != nil {
+			if err == db.ErrEndOfBuildEventStream {
+				break
 			}
+			logger.Error("Syslog drainer getting next event error.", err)
+			return err
+		}
 
-			err = build.SetDrained(true)
+		if ev.Event == "log" {
+			var log event.Log
+
+			err := json.Unmarshal(*ev.Data, &log)
+			if err != nil {
+				logger.Error("Syslog drainer unmarshalling log error.", err)
+				return err
+			}
+
+			payload := log.Payload
+			tag := build.TeamName() + "/" + build.PipelineName() + "/" + build.JobName() + "/" + build.Name() + "/" + string(log.Origin.ID)
+
+
+			err = syslog.Write(d.hostname, tag, time.Unix(log.Time, 0), payload)
 			if err != nil {
-				logger.Error("Syslog drainer setting drained on build error.", err)
+				logger.Error("Syslog drainer sending to server error.", err)
 				return err
 			}
 		}
 	}
+
+	err = build.SetDrained(true)
+	if err != nil {
+		logger.Error("Syslog drainer setting drained on build error.", err)
```

Comment:
Adding to the comments regarding prefixing the error actions, we can also evict the `error` suffix there (as the `log_level` is already telling that this is an error).

Across the codebase, I can recall the use of `failed` sometimes though (`failed-to-send-...` to be explicit that the action of "sending" something failed).

Wdyt?